<launch>
	<node name="rviz" pkg="rviz" type="rviz" respawn="false" output="screen" args=""/>

	<group ns="transforms">
		<node pkg="tf" type="static_transform_publisher" name="base_link_broadcaster" args="0 0 0 0 0 0 base_footprint base_link 100"/>
		<node pkg="tf" type="static_transform_publisher" name="imu_link_broadcaster" args="0.0 0 0.0 0 0 0 base_footprint imu_link 100"/>
	</group>

	<group ns="fmSensors">
		<node pkg="sparkfun_razor_9dof" name="imu_node"  type="sparkfun_9dof_node" output="screen" >
			<param name="nmea_from_imu_sub" value="/fmData/nmea_from_imu"/>
			<param name="imu_pub" value="/fmInformation/imu"/>
			<param name="magnetometer_pub" value="/fmInformation/magnetometer"/>
			<param name="publish_imu" value="true"/>
			<param name="publish_magnetometer" value="true"/>
			<param name="use_enu" value="true"/>
			<param name="frame_id" value="imu_link"/>
		 </node>
	</group>

	<group ns="fmLib">
		<node pkg="serial_nmea" name="imu_nmea_node" type="nmea_node" output="screen">
		    <param name="str_to_msg_sub" value="/fmData/imu_rx" />   
		    <param name="str_to_msg_pub" value="/fmData/nmea_from_imu" />
		    <param name="msg_to_str_sub" value="/fmData/nmea_to_imu" />   
		    <param name="msg_to_str_pub" value="/fmData/imu_tx" />           
		    <!-- <param name="use_nmea_checksum" type="boolean" value="true" /> -->
		</node>

		<node pkg="serial_string" name="imu_serial_string_node" type="serial_string_node" output="screen" >
			<param name="publisher_topic" value="/fmData/imu_rx" />	 
			<param name="subscriber_topic" value="/fmData/imu_tx" />
			<param name="device" value="/dev/ttyUSB0" />	
			<param name="baudrate" value="57600" />
			<param name="termination_character" value="10"/>
		</node> 	
	</group>
</launch>
